database/connector.py

```diff
@@ -1,34 +1,181 @@
 from typing import NoReturn
+import sqlite3
 
 
 class Connector:
     def __init__(self):
         """
         Establishes connection to database etc.
         """
-        raise NotImplementedError
+        self.conn = sqlite3.connect('database.sqlite')
+        self._create_database()
 
     def _create_database(self) -> NoReturn:
         """
         Create database
         Maybe we need to manually apply the migration mechanism instead of this method
         """
-        raise NotImplementedError
+        cursor = self.conn.cursor()
+        cursor.execute("PRAGMA foreign_keys=on;")
+        cursor.execute("CREATE TABLE IF NOT EXISTS users "
+                       "(id INTEGER PRIMARY KEY AUTOINCREMENT, name VARCHAR, event_id INTEGER, "
+                       "FOREIGN KEY (event_id) REFERENCES events(id), UNIQUE(name, event_id) ON CONFLICT REPLACE);")
+        cursor.execute("CREATE TABLE IF NOT EXISTS events "
+                       "(id INTEGER PRIMARY KEY AUTOINCREMENT, name VARCHAR);")
+        cursor.execute("CREATE TABLE IF NOT EXISTS expenses "
+                       "(id INTEGER PRIMARY KEY AUTOINCREMENT, description VARCHAR, payer VARCHAR, sum INTEGER, "
+                       "FOREIGN KEY (payer) REFERENCES users(id));")
+        cursor.execute("CREATE TABLE IF NOT EXISTS debtors "
+                       "(id INTEGER PRIMARY KEY AUTOINCREMENT, expense_id INTEGER, name INTEGER, sum INTEGER, "
+                       "FOREIGN KEY (expense_id) REFERENCES expenses(id), FOREIGN KEY (name) REFERENCES users(id));")
+        cursor.execute("CREATE TABLE IF NOT EXISTS payments "
+                       "(id INTEGER PRIMARY KEY AUTOINCREMENT, sender INTEGER, recipient INTEGER, sum INTEGER,"
+                       "FOREIGN KEY (sender) REFERENCES users(id), FOREIGN KEY (recipient) REFERENCES users(id));")
+
+    def save_event_info(
+            self,
+            event_name: str):
+        cursor = self.conn.cursor()
+        cursor.execute("INSERT INTO events (name) VALUES(?);", (event_name,))
```

Comment:
Is this method protected from SQL-injection? We need to be sure about that

Reply:
at first glance, yes